### PR TITLE
API key authentication in HTTP server

### PR DESCRIPTION
**Pull request by teogeb**
HTTP server middleware for `apiAuthenticator`. 

If the config lists some API keys in (in `apiAuthentication` block), all requests to httpServer need to be authenticated. The key is provided in the  HTTP header: `Authorization: Bearer <apiKey>`

### CODE REVIEW

File: packages/broker/test/unit/httpServer.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+describe('HttpServer', () => {
+
+    describe('ApiAuthenticator', () => {
+        
+        it('no authentication required', async () => {
+            const server = await startTestServer(null)
+            const response = await createRequest()
+            const body = await response.text()
+            expect(body).toBe('FOO')
+            await stopServer(server)
```
Comment (harbu) [inline packages/broker/test/unit/httpServer.test.ts:38]: `stopServer` might not get invoked if something fails before, leading to subsequent tests in this test suite to all fail. Wrap with `try-finally` or move to `afterEach`?

**Review by harbu — CHANGES_REQUESTED** [review]
LGTM; see comment on test

File: packages/broker/test/unit/httpServer.test.ts (hunk 1 above)
Comment (teogeb) [reply packages/broker/test/unit/httpServer.test.ts:38]: Fixed

File: packages/broker/test/unit/httpServer.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        it('valid authentication', async () => {
+            const server = await startTestServer({
+                keys: [MOCK_API_KEY]
+            })
+            try {
+                const response = await createRequest({
+                    Authorization: `Bearer ${MOCK_API_KEY}`
+                })
+                const body = await response.text()
+                expect(body).toBe('FOO')
```
Comment (harbu) [inline packages/broker/test/unit/httpServer.test.ts:50]: I belive there is a possibility that these `expect` blocks never get executed yet the test will still show green. I think you need to add a catch into these

```ts
try {
  ...
  expect(...).toEqual(...)
} catch (e) {
    fail(`failed due to ${e}`)
} finally {
   await stopServer(server)
}
```

Comment (teogeb) [reply packages/broker/test/unit/httpServer.test.ts]: Refactored to use `afterEach`

File: packages/broker/test/unit/httpServer.test.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    })
+
+    describe('ApiAuthenticator', () => {
+        
+        it('no authentication required', async () => {
+            server = await startTestServer(null)
+            const response = await createRequest()
+            const body = await response.text()
+            expect(body).toBe('FOO')
+            await stopServer(server)
```
Comment (harbu) [inline packages/broker/test/unit/httpServer.test.ts:47]:
```suggestion
```

File: packages/broker/test/unit/httpServer.test.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                keys: [MOCK_API_KEY]
+            })
+            const response = await createRequest({
+                Authorization: 'Bearer invalid-api-key'
+            })
+            expect(response.status).toBe(403)
+        })
+
+        it('unauthorized', async () => {
+            const server = await startTestServer({
```
Comment (harbu) [inline packages/broker/test/unit/httpServer.test.ts:72]:
```suggestion
            server = await startTestServer({
```

**Review by harbu — APPROVED** [review]

**Pull request merged**